Mage.Sets/src/mage/cards/t/TimotharBaronOfBats.java

```diff
@@ -0,0 +1,143 @@
+                                "If you do, create a 1/1 black Bat creature token with flying. " +
+                                "It gains \"When this creature deals combat damage to a player, " +
+                                "sacrifice it and return the exiled card to the battlefield tapped\"."),
+                false,
+                anotherVampireFilter,
+                true
+        ));
+    }
+
+    private TimotharBaronOfBats(final TimotharBaronOfBats card) { super(card); }
+
+    @Override
+    public Card copy() { return new TimotharBaronOfBats(this); }
+}
+
+class TimotharBaronOfBatsCreateBatEffect extends OneShotEffect {
+
+    TimotharBaronOfBatsCreateBatEffect() {
+        super(Outcome.Benefit);
+        staticText = "exile it. " +
+                "If you do, create a 1/1 black Bat creature token with flying. " +
+                "It gains \"When this creature deals combat damage to a player, " +
+                "sacrifice it and return the exiled card to the battlefield tapped\".";
+    }
+
+    private TimotharBaronOfBatsCreateBatEffect(final TimotharBaronOfBatsCreateBatEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) { return false; }
+
+        Card card = game.getCard(targetPointer.getFirst(game, source));
+        if (card == null) { return false; }
+
+        player.moveCards(card, Zone.EXILED, source, game);
+
+        BatToken bat = new BatToken();
+        bat.addAbility(new DealsCombatDamageToAPlayerTriggeredAbility(
+                new TimotharBaronOfBatsReturnEffect(card, game),
+                false
+        ));
+
+        CreateTokenEffect createBatEffect = new CreateTokenEffect(bat, 1);
+        createBatEffect.apply(game, source);
+
+        return false;
+    }
+
+    @Override
+    public TimotharBaronOfBatsCreateBatEffect copy() { return new TimotharBaronOfBatsCreateBatEffect(this); }
+}
+
+class TimotharBaronOfBatsReturnEffect extends OneShotEffect {
+
+    private final MageObjectReference morOfCardToReturn;
+
+    TimotharBaronOfBatsReturnEffect(Card card, Game game) {
+        super(Outcome.PutCreatureInPlay);
+        staticText = "return the exiled card to the battlefield tapped";
+        this.morOfCardToReturn = new MageObjectReference(card, game);
+    }
+
+    private TimotharBaronOfBatsReturnEffect(final TimotharBaronOfBatsReturnEffect effect) {
+        super(effect);
+        this.morOfCardToReturn = effect.morOfCardToReturn;
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) { return false; }
+
+        Card card = morOfCardToReturn.getCard(game);
+        if (card == null) { return false; }
+
+        new SacrificeSourceEffect().apply(game, source);
```

Comment:
Maybe you missed my last comment, but I would have this just be a second effect on the ability.  It would fix the text of the trigger too.

Reply:
Oh it totally didn't, I hit resolve conversation.